src/crimson/os/seastore/extent_placement_manager.cc

```diff
@@ -1043,11 +1039,36 @@ RandomBlockOolWriter::do_write(
     } else {
       ceph_assert("impossible");
     }
+
+    // TODO : allocate a consecutive address based on a transaction
+    if (writes.size() != 0 &&
+       (writes.back().offset + writes.back().bp.length() == paddr &&
+       writes.back().bp.c_str() + writes.back().bp.length() ==
+       ex->get_bptr().c_str())) {
+      // We can write both the currrent extent and the previous one at once
+      // if the extents are located in a raw
+      writes.back().bp = buffer::claim_char(
+	writes.back().bp.length() + ex->get_bptr().length(),
+	writes.back().bp.c_str());
```

Comment:
So this means the life cycle of of `writes.back().bp` is relying on that both extents must be alive during the followup writes, considering the transaction can be invalidated in the middle.

Reply:
Right.